lbz/jwt_utils.py

```diff
@@ -0,0 +1,51 @@
+import json
+import os
+
+from jose import jwt
+from jose.exceptions import JWTError, JWTClaimsError, ExpiredSignatureError
+
+from lbz.exceptions import Unauthorized, ServerError
+from lbz.misc import logger
+
+PUBLIC_KEYS = []
+ALLOWED_AUDIENCES = []
+
+if allowed_pubkeys_str := os.environ.get('ALLOWED_PUBLIC_KEYS'):
+    PUBLIC_KEYS.extend(json.loads(allowed_pubkeys_str)["keys"])
+if allowed_audiences_str := os.environ.get('ALLOWED_AUDIENCES'):
+    ALLOWED_AUDIENCES.extend(allowed_audiences_str.split(','))
+
+
+def get_matching_jwk(auth_jwt_token: str) -> dict:
+    try:
+        kid_from_jwt_header = jwt.get_unverified_header(auth_jwt_token)["kid"]
+        for key in PUBLIC_KEYS:
+            if key["kid"] == kid_from_jwt_header:
+                return key
+
+        logger.error(f"Required key not found in configuration.")
+        raise Unauthorized
+    except (JWTError, KeyError):
```

Comment:
Probably there won't be JWTError now

Reply:
Looks like `get_unverified_header` can still throw this error if the token is invalid